### ISSUE TITLE
The problem caused by hidden imports

### ISSUE THREAD

**Issue by Kukaina** [Nuitka/Nuitka#2967]
When I use nuitka to package a program that includes yt-dlp, its progress gets stuck and it takes a long time (about a night) to show that it is completed. However, the size of the packaged program is abnormally large, over 500M, which should be only about 20M normally. I have already raised an issue with yt-dlp, but this seems to be a problem with nuitka.
The developer of yt-dlp said that `Nuitka isn't correctly detecting yt-dlp's imports. Pyinstaller solves this by allowing a package to specify its "hidden imports" (which is done by yt-dlp here). Nuitka does not seem to have an equivalent to this, and therefore yt-dlp can't support building with Nuitka. I think you'll need to open an issue with Nuitka about this`
The link to the original issue is [this](https://github.com/yt-dlp/yt-dlp/issues/10307#issuecomment-2198761100)

**Comment by KRRT7** [comment]
please provide a minimal reproducible example.

**Comment by Kukaina** [comment]
> please provide a minimal reproducible example.

I created a virtual environment (test1), and only executed pip install nuitka yt-dlp, this is its piplist
![image](https://github.com/Nuitka/Nuitka/assets/149920301/f2ee999c-c45c-475d-948c-fcc72c44bd00)
Then I created a 1.py and wrote the following content in it
```Python
from yt_dlp import YoutubeDL

with YoutubeDL()as ydl:
    ydl.download("https://www.bilibili.com/video/BV14r421F7iJ")
```
Then run `nuitka  --standalone ./1.py`
Then it stuck in the position shown in the picture
![image](https://github.com/Nuitka/Nuitka/assets/149920301/575d7142-b424-4bc4-b538-0f697a0bbea9)

**Comment by kayhayen** [comment]
Scalability issue for all I can tell, but I got something coming for that, maybe it will help this too.

**Comment by bashonly** [comment]
For clarity, my comment about hidden imports didn't have anything to do with the hanging on lazy extractors, it was in reply to what happens when we remove lazy extractors altogether: https://github.com/yt-dlp/yt-dlp/issues/10307#issuecomment-2198479973

**Comment by KRRT7** [comment]
> its progress gets stuck and it takes a long time (about a night) to show that it is completed

this is due to an abnormally large file within yt-dlp (`yt_dlp.extractor.lazy_extractors.py`) which is about 15k lines full of classes

i'm debugging the large binary size now.

**Comment by KRRT7** [comment]
the binary crash has been fixed, when compiled (with a limited `yt_dlp.extractor.lazy_extractors.py` i.e just the Youtube classes, the binary is 80MB, pretty reasonable for what it is. the 500MB size you were getting might be due to the 15k lines worth of class, i'll leave that compiling and see if it lines up

**Comment by bashonly** [comment]
> when compiled (with a limited `yt_dlp.extractor.lazy_extractors.py` i.e just the Youtube classes, 

Just an FYI: this will break yt-dlp functionality for all sites other than Youtube. It would be better to remove the `lazy_extractors.py` file entirely.

**Comment by kayhayen** [comment]
Nah, we are not removing it. There is worry about scalability, but we will have something to make many classes compile much faster in the coming releases, then we need not worry. We try to never break software with anti-bloat.

**Comment by bashonly** [comment]
`lazy_extractors.py` is optional; removing it **does not** break yt-dlp. Removing all classes from `lazy_extractors` besides the Youtube classes **does** break yt-dlp.

**Comment by zak-45** [comment]
Hi, moved from pytube to yt-dlp and got this error now : 
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: cl (cl 14.3).
Unexpected output from this command:
<clcache> /Fomodule.yt_dlp.extractor.lazy_extractors.obj /c "module.yt_dlp.extractor.lazy_extractors.c" /nologo /std:c11 /wd5105 /Ox /GF /Gy /EHsc /J /Gd /bigobj /MT /D__NUITKA_NO_ASSERT__ /D_NUITKA_WINMAIN_ENTRY_POINT /D_NUITKA_STANDALONE /D_NUITKA_ONEFILE_MODE /D_NUITKA_CONSTANTS_FROM_RESOURCE /D_NUITKA_FROZEN=153 /D_NUITKA_EXE /D_NUITKA_PLUGIN_MULTIPROCESSING_ENABLED=1 /IC:\HOSTED~1\windows\Python\311~1.9\x64\Lib\SITE-P~1\nuitka\build\inline_copy\zlib /IC:\HOSTED~1\windows\Python\311~1.9\x64\include /I. /IC:\HOSTED~1\windows\Python\311~1.9\x64\Lib\SITE-P~1\nuitka\build\include /IC:\HOSTED~1\windows\Python\311~1.9\x64\Lib\SITE-P~1\nuitka\build\static_src /IC:\HOSTED~1\windows\Python\311~1.9\x64\Lib\SITE-P~1\nuitka\build\inline_copy\libbacktrace
D:\a\WLEDVideoSync\WLEDVideoSync\build\WLEDVideoSync.build\module.yt_dlp.extractor.lazy_extractors.c(36800) : fatal error C1002: compiler is out of heap space in pass 2

Nuitka v: 2.4.4

More info can be found there : 
https://github.com/zak-45/WLEDVideoSync/actions/runs/10181533151/job/28161925789

Reported here looks like same problem.

**Comment by kayhayen** [comment]
For low memory issues check the documentation on the website.
